### PR TITLE
Symfony 7 compatibillity

### LINKED ISSUE #630: Symfony 7 compatibility
**Is your feature request related to a problem? Please describe.**
No upgrade to SF 7 possible

**Describe the solution you'd like**
Upgrade to SF 7

**Describe alternatives you've considered**
Downgrade to SF 6.4

Thanks

### PR DESCRIPTION
**Pull request by kl3sk**
This may allow Symfony 7 support.

Older version were dropped and older PHP version.

Fell free to comment/annotate 😃 

Fixes #630

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,21 +10,18 @@ jobs:
...
+        symfony-version: ['^6.4']
         include:
-          - php-version: 7.4
-            symfony-version: "^4.4"
-            composer-flags: "--prefer-lowest"
-          - php-version: 7.4
-            symfony-version: "^4.4"
-          - php-version: 8.0
-            symfony-version: "^4.4"
           - php-version: 8.1
```
Comment (alexislefebvre) [inline .github/workflows/tests.yml:19]:
```suggestion
          - php-version: 7.4
            symfony-version: "^5.4"
            composer-flags: "--prefer-lowest"
          - php-version: 8.1
            symfony-version: "^5.4"
          - php-version: 8.1
```

Let's keep the tests with Symfony 5.4.

**Review by alexislefebvre — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kl3sk) [reply .github/workflows/tests.yml:19]: Ok, I was pretty sure you'll say something like this 👍

**Comment by alexislefebvre** [comment]
Do you know why it fails with PHP 8.2? It may be due to the fact that annotations are not working with PHP 8.2+. :grimacing:

**Comment by kl3sk** [comment]
This what I was searching for, and agree that is a Doctrine Annotation vs Attributes conflict.

If I'm not wrong, both can be used at the same time.

I'll have a try

**Comment by alexislefebvre** [comment]
Please hold on, let's check something first: https://github.com/liip/LiipFunctionalTestBundle/issues/630#issuecomment-1878887565

**Comment by kl3sk** [comment]
Ok, so what's next.

Do we close it or I can push my tries with attributes to see what happens ?

**Comment by alexislefebvre** [comment]
As you want, you can try to migrate to a YAML configuration as in LiipFunctionalTestBundle: https://github.com/liip/LiipTestFixturesBundle/blob/19c9f4281bff0d5e3571de4ba71b66c85ba0b0c4/tests/App/DataFixtures/ORM/user.yml

Otherwise we'll be able to do it, no problem.

**Comment by kl3sk** [comment]
`LiipFunctionalTestBundle` Works fine, this bundle may be at least upgradable to 7.0

The other problème is here I think.

https://github.com/liip/LiipFunctionalTestBundle/blob/5658b8a2781e12c2d195a83111e171d659f1fdf9/tests/App/config.yml#L39

But how say use attribute for 8.2 and annotation otherwise

I change to attribute to test it.

**Comment by alexislefebvre** [comment]
> But how say use attribute for 8.2 and annotation otherwise

Are annotations still necessary? I'm pretty sure that attributes can be used with Symfony 5.4+.

**Comment by alexislefebvre** [comment]
If you want to continue to work on this subject, please open another PR where you only drop support of Symfony 4.4 (you'll have to create a branch for that). Once it will be merged, you'll be able to rebase this PR and tests will be run automatically.

File: composer.json [hunk 2]
```diff
@@ -15,27 +15,27 @@
         }
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^7.4 || ^8.0 || ^8.1 || ^8.2",
```
Comment (Chris53897) [inline composer.json:18]: Please revert "^7.4 || ^8.0" is fine

File: tests/App/Entity/User.php [hunk 3]
```diff
@@ -22,6 +22,10 @@
  * @ORM\Entity()
  * @ORM\Table("liip_user")
  */
+#[
```
Comment (Chris53897) [inline tests/App/Entity/User.php:25]: Can you change this, to the following, and see if this fix some errors? On Php 7.4 run.

#[ORM\Entity()]
#[ORM\Table(name: "liip_user")]

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -10,21 +10,23 @@ jobs:
...
-            symfony-version: "^6.2"
+            symfony-version: "^5.4"
+          - php-version: 8.1
+            symfony-version: "^6.4"
+          - php-version: 8.2
+            symfony-version: "^6.4"
           - php-version: 8.2
```
Comment (Chris53897) [inline .github/workflows/tests.yml:28]: Can you please add a testrun for php 8.3 as well?

**Comment by alexislefebvre** [comment]
> > But how say use attribute for 8.2 and annotation otherwise
> 
> Are annotations still necessary? I'm pretty sure that attributes can be used with Symfony 5.4+.

My bad, it requires PHP 8.0+. :grimacing:

**Comment by alexislefebvre** [comment]
I opened a new PR based on this one:

- GH-634

**Comment by alexislefebvre** [comment]
See the release: https://github.com/liip/LiipFunctionalTestBundle/issues/630#issuecomment-1880127819

File: .github/workflows/tests.yml (hunk 4 above)
Comment (alexislefebvre) [reply .github/workflows/tests.yml:28]: Feel free to add it in a new PR.

**Pull request merged**